### ISSUE TITLE
S3260 does not detect if the class is inherited by a sub-class

### ISSUE THREAD

**Issue by jairbubbles** [SonarSource/sonar-dotnet#4983]
### Description

S3260 does not detect if the class is inherited by a sub-class.

### Repro steps

Just paste this example:

```csharp
internal class Test
{
    private class PropertyFetch // S3260 Private classes or records which are not derived in the current assembly should be marked as 'sealed'.
    {
        public virtual bool Fetch() => false;

        private sealed class CustomPropertyFetch : PropertyFetch
        {
            public override bool Fetch() => true;
        }
    }
}
```
And observe that S3260 is displayed while it shouldn't. 
Anyway the class can't be sealed as it has a virtual method.

### Known workarounds

Set the class as abstract (might not be applicable).

### Related information

```xml
<PackageReference Include="SonarAnalyzer.CSharp" Version="8.30.0.37606" />
```
